pkg/reconciler/service/service_test.go

```diff
@@ -1382,6 +1382,26 @@ func TestReconcile(t *testing.T) {
 		WantServiceReadyStats: map[string]int{
 			"foo/new-owner": 1,
 		},
+	}, {
+		Name: "service is not updated but without latest defaults; config and route should not be updated",
+		Objects: []runtime.Object{
+			DefaultService("release-no-change", "foo", WithRunLatestRollout, WithInitSvcConditions, func(service *v1alpha1.Service) {
+				if service.Spec.ConfigurationSpec.Template == nil {
+					service.Spec.ConfigurationSpec.Template = &v1alpha1.RevisionTemplateSpec{}
+				}
+				service.Spec.ConfigurationSpec.Template.Annotations = map[string]string{}
+			}),
+			config("release-no-change", "foo", WithRunLatestRollout),
+			route("release-no-change", "foo", WithRunLatestRollout),
```

Comment:
looks like `config()` and `route()` both get resources with `defaulting` already applied to them. so this wont really exercise the semantic equality behavior intended